### DISCUSSION TITLE
Builtin Desktop automatic shutdown

### DISCUSSION THREAD

**Discussion by ghost** [Ylianst/MeshCentral#5815] [General]
Hi all,

I'm using the Mesh for getting to desktops on other devices and I'm using the  _"userSessionIdleTimeout"_ to logout automaticlly.
I've noticed that the Desktop is not shuting down upon timeout limit.
I'm getting a prompt in the browser asking me if I want to leave the site.
I've tried the  "agentIdleTimeout" but it's not shuting down the Desktop and it doesn't seem to do anything at all.
Is there a way to force the Desktop to shutdown upon user idle timeout?
Thanks.

**Comment by si458** [comment]
`userSessionIdleTimeout` is the timeout value for the web client, not the remote computers desktop?
This is shown by it popping up asking to navigate away from this page.
I think the might be a bug, as it should disconnect the remote desktop session if ones activate then take u bk to the login screen

**Reply by ghost** [reply]
userSessionIdleTimeout is doing that but if you're connected to another device with the Deskop, the browser will ask if you want to to leave or stay.
Moreover, you need to click twice on the confirmation wether you want to stay or leave. 
Also, what is the difference between "userSessionIdleTimeout" and "agentIdleTimeout"?

**Reply by si458** [reply]
If you read the schema file, `agentIdleTimeout - How much time in seconds with no traffic from an agent before dropping the agent connection.` so best to not set this as this is for remote devices, not users!

Also this is possibly a bug then, so best to open a github issue with all the details in, and we can look into it, plz follow the template!
